### ISSUE TITLE
Couldn't compile kernel (Windows 10)

### ISSUE THREAD

**Issue by jecs** [JuliaGPU/GPUArrays.jl#35]
All test pass:
```
julia> Pkg.test("GPUArrays")
INFO: Computing test dependencies for GPUArrays...
INFO: No packages to install, update or remove
INFO: Testing GPUArrays
Test Summary: | Pass  Total
julia         |   48     48
Test Summary: | Pass  Total
opencl        |   51     51
Test Summary: | Pass  Total
BLAS          |    8      8
Test Summary: | Pass  Total
Shared        |    6      6
INFO: GPUArrays tests passed
INFO: No packages to install, update or remove
```
Code:
```
julia> x = GPUArray(ones(100,100,100)); y = GPUArray(randn(100,100,100)); z = x .* y;
Couldn't compile kernel:
    1   : // dependant type declarations
    2   : typedef struct {
    3   :     int field1;
    4   :     int field2;
    5   :     int field3;
    6   : }int[3];
    7   :
    8   : typedef struct {
    9   : float empty; // structs can't be empty
    10  : }Base123;
    11  :
    12  : // dependant function declarations
    13  : float broadcast_index_1(__global float * restrict  arg, int[3] shape, int i)
    14  : {
    15  :     ;
    16  :     return arg[i - 1];
    17  :     ;
    18  : }
    19  : // Main inner function
    20  : __kernel void broadcast_kernel_2(__global float * restrict  A, Base123 f, int[3] sz, __global float * restrict  arg_1, __global float * restrict  arg_2)
    21  : {
    22  :     int i;
    23  :     i = get_global_id(0) + 1;
    24  :     float _ssavalue_0;
    25  :     _ssavalue_0 = broadcast_index_1(arg_1, sz, i) * broadcast_index_1(arg_2, sz, i);
    26  :     A[i - 1] = _ssavalue_0;
    27  :     ;
    28  : }
    29  :
With following build error:
<kernel>:6:2: error: expected ';' after struct
}int[3];
 ^
 ;
<kernel>:2:1: warning: typedef requires a name
typedef struct {
^~~~~~~
<kernel>:6:5: error: expected identifier or '('
}int[3];
    ^
<kernel>:13:64: error: expected ')'
float broadcast_index_1(__global float * restrict  arg, int[3] shape, int i)
                                                               ^
<kernel>:13:24: note: to match this '('
float broadcast_index_1(__global float * restrict  arg, int[3] shape, int i)
                       ^
<kernel>:13:60: error: parameter name omitted
float broadcast_index_1(__global float * restrict  arg, int[3] shape, int i)
                                                           ^
<kernel>:16:16: error: use of undeclared identifier 'i'
    return arg[i - 1];
               ^
<kernel>:20:82: error: expected ')'
__kernel void broadcast_kernel_2(__global float * restrict  A, Base123 f, int[3] sz, __global float * restrict  arg_1, __global float * restrict  arg_2)
                                                                                 ^
<kernel>:20:33: note: to match this '('
__kernel void broadcast_kernel_2(__global float * restrict  A, Base123 f, int[3] sz, __global float * restrict  arg_1, __global float * restrict  arg_2)
                                ^
<kernel>:20:78: error: parameter name omitted
__kernel void broadcast_kernel_2(__global float * restrict  A, Base123 f, int[3] sz, __global float * restrict  arg_1, __global float * restrict  arg_2)
                                                                             ^
<kernel>:25:37: error: use of undeclared identifier 'arg_1'
    _ssavalue_0 = broadcast_index_1(arg_1, sz, i) * broadcast_index_1(arg_2, sz, i);
                                    ^
<kernel>:25:71: error: use of undeclared identifier 'arg_2'
    _ssavalue_0 = broadcast_index_1(arg_1, sz, i) * broadcast_index_1(arg_2, sz, i);
                                                                      ^
�
ERROR: CLError(code=-11, CL_BUILD_PROGRAM_FAILURE)
Stacktrace:
 [1] macro expansion at C:\Users\jecs\.julia\v0.6\OpenCL\src\macros.jl:6 [inlined]
 [2] #build!#113(::String, ::Bool, ::Function, ::OpenCL.cl.Program) at C:\Users\jecs\.julia\v0.6\OpenCL\src\program.jl:101
 [3] (::OpenCL.cl.#kw##build!)(::Array{Any,1}, ::OpenCL.cl.#build!, ::OpenCL.cl.Program) at .\<missing>:0
 [4] (::Transpiler.CLTranspiler.##27#28{Tuple{GPUArrays.GPUArray{Float64,3,OpenCL.cl.Buffer{Float64},GPUArrays.CLBackend.CLContext},Base.#*,Tuple{Int32,Int32,Int32},GPUArrays.GPUArray{Float64,3,OpenCL.cl.Buffer{Float64},GPUArrays.CLBackend.CLContext},GPUArrays.GPUArray{Float64,3,OpenCL.cl.Buffer{Float64},GPUArrays.CLBackend.CLContext}},GPUArrays.CLBackend.#broadcast_kernel,OpenCL.cl.CmdQueue,OpenCL.cl.Context,NTuple{5,DataType}})() at C:\Users\jecs\.julia\v0.6\Transpiler\src\clike/opencl\compilation.jl:104
 [5] get!(::Transpiler.CLTranspiler.##27#28{Tuple{GPUArrays.GPUArray{Float64,3,OpenCL.cl.Buffer{Float64},GPUArrays.CLBackend.CLContext},Base.#*,Tuple{Int32,Int32,Int32},GPUArrays.GPUArray{Float64,3,OpenCL.cl.Buffer{Float64},GPUArrays.CLBackend.CLContext},GPUArrays.GPUArray{Float64,3,OpenCL.cl.Buffer{Float64},GPUArrays.CLBackend.CLContext}},GPUArrays.CLBackend.#broadcast_kernel,OpenCL.cl.CmdQueue,OpenCL.cl.Context,NTuple{5,DataType}}, ::Dict{Any,Transpiler.CLTranspiler.CLFunction}, ::Tuple{GPUArrays.CLBackend.#broadcast_kernel,NTuple{5,DataType}}) at .\dict.jl:449
 [6] Transpiler.CLTranspiler.CLFunction(::Function, ::Tuple{GPUArrays.GPUArray{Float64,3,OpenCL.cl.Buffer{Float64},GPUArrays.CLBackend.CLContext},Base.#*,Tuple{Int32,Int32,Int32},GPUArrays.GPUArray{Float64,3,OpenCL.cl.Buffer{Float64},GPUArrays.CLBackend.CLContext},GPUArrays.GPUArray{Float64,3,OpenCL.cl.Buffer{Float64},GPUArrays.CLBackend.CLContext}}, ::OpenCL.cl.CmdQueue) at C:\Users\jecs\.julia\v0.6\Transpiler\src\clike/opencl\compilation.jl:79
 [7] acc_broadcast!(::Base.#*, ::GPUArrays.GPUArray{Float64,3,OpenCL.cl.Buffer{Float64},GPUArrays.CLBackend.CLContext}, ::Tuple{GPUArrays.GPUArray{Float64,3,OpenCL.cl.Buffer{Float64},GPUArrays.CLBackend.CLContext},GPUArrays.GPUArray{Float64,3,OpenCL.cl.Buffer{Float64},GPUArrays.CLBackend.CLContext}}) at C:\Users\jecs\.julia\v0.6\GPUArrays\src\backends\opencl\opencl.jl:168
 [8] broadcast(::Function, ::GPUArrays.GPUArray{Float64,3,OpenCL.cl.Buffer{Float64},GPUArrays.CLBackend.CLContext}, ::GPUArrays.GPUArray{Float64,3,OpenCL.cl.Buffer{Float64},GPUArrays.CLBackend.CLContext}) at C:\Users\jecs\.julia\v0.6\GPUArrays\src\abstractarray.jl:196
```
Version info:
```
Julia Version 0.6.0
Commit 903644385b* (2017-06-19 13:05 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.1 (ORCJIT, skylake)
```
GPUArrays.jl status:
```
julia> Pkg.status("GPUArrays")
 - GPUArrays                     0.0.2+             master
```

**Comment by SimonDanisch** [comment]
I'm working on it at the moment :) let me see if I can merge things before monday

**Comment by SimonDanisch** [comment]
Solved by #37
